### ISSUE TITLE
Can't view a dashboard without data access permission for fields connected to filters

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#10656]
**Describe the bug**
A user with collection permissions to a dashboard _also_ needs data access permission 

**Logs**
```
08-20 18:06:11 WARN middleware.log :: GET /api/field/21 403 1.9 ms (4 DB calls) 
"You don't have permissions to do that."

08-20 18:06:11 WARN middleware.log :: GET /api/field/21/values 403 1.2 ms (3 DB calls) 
"You don't have permissions to do that."
```

**To Reproduce**
1. Create a dashboard
2. Create a "count of rows" card for the orders dataset
3. Add this card to a dashboard
4. Add an ID parameter and link it to the User ID field
3. Create a user X 
4. Remove Data access permissions from All Users for Sample Dataset's  People table.
5. Login as user X and go to the dashboard you created 

**Expected behavior**
I should be able to use a dashboard I have collections access to even if I have no self-service data access permissions 

**Information about your Metabase Installation:**
- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
Chrome
- Your operating system: (e.x. OS X 10, Windows XP, etc)
Mac OS X
- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
Sample dataset
- Metabase version: (e.x. 0.19.3)
```
You're on version v0.34.0-snapshot
Built on 2019-08-20
Hash: 3d5bd0a
```

- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Linux/Ubuntu 12)
Macbook

- Metabase internal database: (e.x. H2 (default), MySQL, Postgres)
H2

**Comment by walterl** [comment]
I've hit the same issue, but it gets more interesting: I have a dashboard containing 2 cards, each for a different database. The logged in user has collection access, and data+SQL permissions on one the database. When navigating to the dashboard, just a permissions error is rendered:

![Screenshot 2019-08-22 01:12:16](https://user-images.githubusercontent.com/23798/63475389-e8aab080-c47c-11e9-82c5-ee63677f936e.png)

Clicking the browser's _back_ button (to `collection/root`) and then selecting the dashboard again, causes the dashboard to render:

![Screenshot 2019-08-22 01:14:05](https://user-images.githubusercontent.com/23798/63475428-1c85d600-c47d-11e9-83cb-32df0db366a5.png)

In both cases the appropriate 403 responses from the back-end can be seen in the request logs.

**Comment by walterl** [comment]
Here are the logs for the two cases:

* Browser refreshed at collections screen, and dashboard selected: [refreshed.log](https://github.com/metabase/metabase/files/3527738/refreshed.log)
* Going _back_ and then selecting dashboard from collection: [back-n-fwd.log](https://github.com/metabase/metabase/files/3527737/back-n-fwd.log)

**Comment by camsaul** [comment]
Fixed by #10655
